src/pokemon/DecompressTask.java

```diff
@@ -115,49 +140,75 @@ private int getSpriteSize() throws IOException {
 	}
 
 	private void readToBuffer(Sprite sprite) throws IOException {
-		byte initialPacket = this.reader.next();
+		byte initialPacket = this.reader.next(); bitsRead++;
+		println("Initial packet = %s", initialPacket == 0 ? "RLE" : "DATA");
+
 		boolean doingRLE = initialPacket == 0;
 
 		int bitsInSprite = sprite.getAbsoluteWidth() * sprite.getAbsoluteHeight();
 
-		int bitsRead = 0;
+		int rleCount = 0;
+		int dataCount = 0;
+		int bitsReadLocally = 0;
 		int x = 0;
 		int y = 0;
 
-		while (bitsRead <= bitsInSprite) {
-			byte[] bitPairs;
-
-			try {
-				bitPairs = doingRLE ? RLE(bitsRead, bitsInSprite) : DATA(bitsRead, bitsInSprite);
-			} catch (EOFException e) {
-				System.out.println(bitsRead);
-				break;
-			}
+		try {
+			while (bitsReadLocally < bitsInSprite) {
```

Comment:
FYI, this is the root fix for this class. The rest of the stuff in here is diagnostic or debug.